File: redsheet-portal/src/app/database-models/lock-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class LockRecord {
    constructor(lockRecord?: LockRecord )  {
        let properties = [ "applicationModuleId", "childModuleName", "expiresAt", "lockRecordId", "recordId", "userName",];
        for (let property of properties)
            if (lockRecord && lockRecord[property])
                this[property] = lockRecord[property];
    }
 
    @range(1,2147483647)
	applicationModuleId : number =   undefined;
 
    @maxLength(100)
	childModuleName : string =   undefined;
 
    @required()
	expiresAt : Date =   undefined;
 
	lockRecordId : number =   0 ;
 
    @range(1,2147483647)
	recordId : number =   undefined;
 
    @required()
    @maxLength(100)
	userName : string =   undefined;


}

File: RedSheet.DbEntities/Models/LockRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("LockRecords", Schema = "dbo")]
	[Serializable]
	public partial class LockRecord  
	{
 
	[Range(1, int.MaxValue)]
	public int ApplicationModuleId { get; set; } 
 
	[MaxLength(100)]
	public string ChildModuleName { get; set; } 
 
	[Required]
	public System.DateTime ExpiresAt { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int LockRecordId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int RecordId { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string UserName { get; set; } 

	public LockRecord()
	{
    }

    }
}

